Corgibytes.Freshli.Cli/Program.cs

```diff
@@ -24,25 +31,66 @@ public static async Task<int> Main(string[] args)
                 .InvokeAsync(args);
         }
 
+        private static void ConfigureLogging(string consoleLogLevel, string logfile)
+        {
+            NLog.LogLevel desiredLevel = NLog.LogLevel.FromString(consoleLogLevel);
+
+            LoggingConfiguration config = new LoggingConfiguration();
+            if (IsNullOrEmpty(logfile))
+            {
+                ColoredConsoleTarget consoleTarget = new ColoredConsoleTarget();
+                config.AddTarget("console", consoleTarget);
+                consoleTarget.Layout = DefaultLogLayout;
+                config.LoggingRules.Add(new LoggingRule("*", desiredLevel, consoleTarget));
+            }
+            else
+            {
+                FileTarget fileTarget = new FileTarget();
+                config.AddTarget("file", fileTarget);
+                fileTarget.FileName = logfile;
+                fileTarget.Layout = DefaultLogLayout;
+                config.LoggingRules.Add(new LoggingRule("*", desiredLevel, fileTarget));
+            }
+
+            LogManager.Configuration = config;
+        }
+
         static IHostBuilder CreateHostBuilder(string[] args) =>
             Host.CreateDefaultBuilder(args)
-            .ConfigureServices((_, services) =>
+                .ConfigureLogging(logging =>
+                {
+                    logging.ClearProviders();
+                    logging.AddNLog();
+                })
+                .UseNLog()
+                .ConfigureServices((_, services) =>
             {
                 new FreshliServiceBuilder(services).Register();
             });
 
         public static CommandLineBuilder CreateCommandLineBuilder()
         {
+            Option logLevelOption = new Option<string>(alias: "--loglevel", description:"the minimum log level to log to console",
+                getDefaultValue: () => "Warn");
+            Option logFileOption = new Option<string?>("--logfile", "file for logs instead of logging to console");
```

Comment:
I think the `logLevelOption` and `logFileOption` should be classes in the `CommandOptions/` folder.  Like the `ScanCommandOptions` class.  Can you check with @mfcorgi?

Reply:
The `ScanCommandOptions` class is used by the parameter binding functionality in `System.CommandLine` to pull the parameter values out of the parsed string and store them in objects. We don't have the opportunity to do that binding here (at least I don't think we do). Instead the values are being looked up dynamically. All of that to say that I don't think it makes sense to create a separate `CommandOptions` class for `MainCommand`.

Reply:
What I did do was move some of this setup into the `MainCommand` class. That helps consolidate that logic and make this chunk of code a little more readable.